pandas/tests/groupby/test_apply.py

```diff
@@ -102,9 +102,89 @@ def f(g):
     group_keys = grouper._get_group_keys()
 
     values, mutated = splitter.fast_apply(f, group_keys)
+
     assert not mutated
 
 
+@pytest.mark.parametrize(
+    "df,group_names",
+    [
+        (
```

Comment:
Nitpicking but I find the readability here difficult. Can you start each set of parametrization with `(DataFrame(` and line break from there as required?